buildpack/detect_test.go

```diff
@@ -21,302 +22,357 @@ import (
 )
 
 func TestDetect(t *testing.T) {
-	spec.Run(t, "Detect", testDetect, spec.Report(report.Terminal{}))
+	bpPath := filepath.Join("testdata", "by-id", "A", "v1", "buildpack.toml")
+	extPath := filepath.Join("testdata", "by-id", "extA", "v1", "extension.toml")
+	for _, path := range []string{bpPath, extPath} {
+		spec.Run(t, fmt.Sprintf("Detect-%s", filepath.Base(path)), testDetect(path), spec.Report(report.Terminal{}))
+	}
```

Comment:
This will run the test suite once for a buildpack and once for an extension. The diff is a bit hard to decipher but may be easier in an IDE.